lib/src/st2110/pipeline/st22_pipeline_tx.c

```diff
@@ -593,12 +593,7 @@ static void tx_st22p_framebuffs_flush(struct st22p_tx_ctx* ctx) {
 
     while (1) {
       if (framebuff->stat == ST22P_TX_FRAME_FREE) break;
-      if (framebuff->stat == ST22P_TX_FRAME_IN_TRANSMITTING) {
-        /* make sure transport to finish the transmit */
-        /* WA to use sleep here, todo: add a transport API to query the stat */
-        mt_sleep_ms(50);
-        break;
-      }
+      if (framebuff->stat == ST22P_TX_FRAME_IN_TRANSMITTING) mt_sleep_ms(50);
```

Comment:
Can we just remove `mt_sleep_ms(50);` also? It has `mt_sleep_ms(10);` later in line 606

Reply:
I also thought about that i think this is solid sugestion 
i will retest it and implement it

Reply:
Nope the status changes due to frambuffer being in the "FREE" state before encoding make it jump of this function 
I think generally the best way will be adding the frame buffer status and updating the architecture to end up on this status 

For now leaving this as WA